proof.go

```diff
@@ -141,6 +142,61 @@ func (proof Proof) IsEmptyProof() bool {
 	return proof.start == proof.end && len(proof.nodes) == 0 && len(proof.leafHash) == 0
 }
 
+func (proof Proof) isValidEmptyRangeProof(nth *NmtHasher, nID namespace.ID, root []byte, leaves [][]byte, checkNS bool) bool {
```

Comment:
1. Could use unit tests
2. Could use GoDoc
3. What is an empty range proof?

Reply:
@rootulp this code was there before. i just pulled it out to a un-exported method for now. as @vgonkivs mentioned in his previous comment, empty proofs does not make sense. i will leave it here as is for now to make it backwards compatible and refactor this in future. will that work?

Reply:
https://github.com/celestiaorg/nmt/issues/292